enterprise_access/apps/api_client/license_manager_client.py

```diff
@@ -62,3 +63,74 @@ def assign_licenses(self, user_emails, subscription_uuid):
         except requests.exceptions.HTTPError as exc:
             logger.exception(exc)
             raise
+
+
+class LicenseManagerUserApiClient(BaseUserApiClient):
+    """
+    API client for calls to the license-manager service. This client is used for user-specific calls,
+    passing the original Authorization header from the originating request.
+    """
+
+    api_base_url = f"{settings.LICENSE_MANAGER_URL}/api/v1/"
+    learner_licenses_endpoint = f"{api_base_url}learner-licenses/"
+    license_activation_endpoint = f"{api_base_url}license-activation/"
+
+    def auto_apply_license_endpoint(self, customer_agreement_uuid):
+        return f"{self.api_base_url}customer-agreement/{customer_agreement_uuid}/auto-apply/"
+
+    def get_subscription_licenses_for_learner(self, enterprise_customer_uuid):
+        """
+        Get subscription licenses for a learner.
+
+        Arguments:
+            enterprise_customer_uuid (str): UUID of the enterprise customer
+        Returns:
+            dict: Dictionary representation of json returned from API
+        """
+        query_params = {
+            'enterprise_customer_uuid': enterprise_customer_uuid,
+        }
+        url = self.learner_licenses_endpoint
+        try:
+            response = self.get(url, params=query_params, timeout=settings.LICENSE_MANAGER_CLIENT_TIMEOUT)
+            return response.json()
+        except requests.exceptions.HTTPError as exc:
+            logger.exception(f"Failed to get subscription licenses for learner: {exc}")
+            raise
+
+    def activate_license(self, activation_key):
+        """
+        Activate a license.
+
+        Arguments:
+            license_uuid (str): UUID of the license to activate
+        """
+        try:
+            url = self.license_activation_endpoint
+            query_params = {
+                'activation_key': activation_key,
+            }
+            response = self.post(url, params=query_params, timeout=settings.LICENSE_MANAGER_CLIENT_TIMEOUT)
+            response.raise_for_status()
+            if response.status_code == 204:  # Response contains no content
+                return None
```

Comment:
nit: this API was updated to no longer return the 204 status code, in favor of a 200 status code with the response body of the now-activated subscription license. I.e., we may not need this check anymore.

See https://github.com/openedx/license-manager/pull/722